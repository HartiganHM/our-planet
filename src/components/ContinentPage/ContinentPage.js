import React from 'react';
import './ContinentPage.css';

const ContinentPage = () => {
  return <div className="ContinentPage">I am an ContinentPage</div>;
};

export default ContinentPage;
