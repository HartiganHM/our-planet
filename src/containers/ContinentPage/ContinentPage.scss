//Layout, Box, Visual, Typography, Misc
@import '../../styles/mixins.scss';
@import '../../styles/variables.scss';
.ContinentPage {
  display: flex;
  flex-direction: column;
  @include width(160);
  @include padding(2);
  margin: auto;
  .continent-name {
    @include contentHeader();
    background-image: url('../../images/branding/Our-Planet-Light-Logo.png');
  }
  .continent-data-container {
    @include contentDataContainer();
    background-image: url('../../images/branding/OP-Logo-Light-Blue.png');
    .wrapper {
      display: flex;
      padding-bottom: $bottom-padding;
      .continent-image {
        @include contentImage();
      }
      .continent-bio {
        flex: 1;
        @include flexColumnCenter();
        .bio-title {
          @include contentDetailTitleAlternate();
        }
        .bio-body {
          @include contentDetailBody();
        }
      }
    }
  }
}

@media screen and (max-width: 1599px) {
  .ContinentPage {
    @include width(128);
  }
}

@media screen and (max-width: 1279px) {
  .ContinentPage {
    @include width(96);
  }
}

@media screen and (max-width: 959px) {
  .ContinentPage {
    @include width(64);
  }
}

@media screen and (max-width: 639px) {
  .ContinentPage {
    @include width(32);
  }
}
