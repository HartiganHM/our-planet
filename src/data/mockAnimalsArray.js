/*eslint-disable*/
export default [
  {
    name: 'Arctic Fox',
    status: 'Least Concern',
    population: 'Several Hundred Thousand',
    scientific_name: 'Vulpes lagopus',
    height: '11 inches',
    weight: '3 to 20 pounds',
    length: '18 to 27 inches',
    habitat: 'Antarctica / Arctic',
    facts:
      'The Arctic fox is primarily a carnivore that lives inland, away from the coasts. They are dependent on the presence of smaller animals (most often lemmings) to survive.  Arctic foxes also hunt for sea birds, fish, and other marine life. Smaller rodent populations waver between times of abundance and scarcity, which leaves the Arctic fox vulnerable when these creatures are low in numbers.',
    human_benefit:
      'The protection of the Arctic fox ensures the safety of a variety of other wildlife in the Arctic region.',
    image: 'arctic fox',
    display: true
  },
  {
    name: 'Arctic Wolf',
    status: 'Least Concern',
    population: '',
    scientific_name: 'Vulpes lagopus',
    height: '',
    weight: '70 to 175 pounds',
    length: '3.2 to 5.9 feet',
    habitat: 'Antarctica / Arctic',
    facts:
      'Often called the "polar wolf" or "white wolf," Arctic wolves inhabit the Arctic regions of North America and Greenland. Thanks to its isolation, the Arctic wolf is not threatened by hunting and habitat destruction in the same way as its southern relatives.',
    human_benefit:
      'Arctic wolves are carnivorous hunters. By nature they help to control the populations of other animals in the region like the musk ox, caribou and Arctic hares.',
    image: 'arctic wolf',
    display: true
  }
];
