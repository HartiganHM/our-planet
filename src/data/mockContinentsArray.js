/*eslint-disable*/
export default [
  {
    id: 1,
    name: 'Africa',
    bio:
      "Africa straddles the equator which bestows a rich diversity of wildlife and landscapes on the world's second largest continent. Large predators roam the sweeping savannas among huge herds of grazing herbivores, while apes, monkeys and snakes inhabit its dark, dense forests. Widely regarded as the place where human life originated, Africa is home to many of the world's best loved and most fascinating animals, as well as to some of its most endangered. Yet it's also home to many species that are as familiar to us in the UK as they are to Africans.",
    image_url:
      'https://our-planet.s3-us-west-2.amazonaws.com/continents/africa.jpg',
  },
  {
    id: 2,
    name: 'Antarctica / Arctic',
    bio: `Antarctica holds the record for being the coldest place on Earth, which together with extensive ice cover and lack of winter sunlight makes the continent pretty inhospitable to life. The sea, rather than the land, is what supports most creatures such as penguins and seals. It's often warmer than the land, so is a good place to retreat to and warm up to a balmy -2 Celsius. Permanently land-living plants and animals in Antarctica are small and ground-hugging to avoid wind chill and the accompanying dehydration.
  The Arctic ice cap first formed some three million years ago, providing a new habitat - polar pack ice - for mammals and birds to exploit. Seals and polar bears are now the primary inhabitants of this sea ice. The bears travel over it and the seals under, meeting at the seals\' breathing holes, to the detriment of the seals! On land, plants tough enough to survive the Arctic cold form the habitat known as tundra. Because of the harsh climate and prolonged darkness, many animals migrate away from the dark Arctic for the winter, returning to take advantage of the 24 hour summer sunshine to breed.`,
    image_url:
      'https://our-planet.s3-us-west-2.amazonaws.com/continents/antarctica+arctic.jpg',
  },
];
