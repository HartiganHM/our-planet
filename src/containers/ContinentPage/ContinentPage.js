import React, { Component } from 'react';
import { connect } from 'react-redux';
import matchObjectInStore from '../../helpers/matchObjectInStore/matchObjectInStore';
import './ContinentPage.css';

export class ContinentPage extends Component {
  constructor() {
    super();
    this.state = {
      continentData: {},
      animals: []
    };
  }

  componentWillMount() {
    if (this.props.continents.length) {
      const continentData = matchObjectInStore(
        this.props,
        'continent',
        'continents'
      );
      const animals = this.matchContinentAnimals(continentData.id);

      this.setState({ continentData, animals });
    }
  }

  componentWillReceiveProps(nextProps) {
    console.log(nextProps.continents)
    const continentData = matchObjectInStore(
      this.props,
      'continent',
      'continents'
    );
    const animals = this.matchContinentAnimals(continentData.id);

    this.setState({ continentData, animals });
  }

  matchContinent = props => {
    const linkedContinent = props.match.params.continent;
    const continentData = props.continents.find(
      continent => continent.name === linkedContinent
    );
    continentData.image = continentData.name.toLowerCase();

    this.setState({ continentData });
  };

  matchContinentAnimals = continentId => {
    const animals = this.props.animals.filter(
      animal => animal.continent_id === continentId
    );
    return animals;
  };

  render() {
    return <div className="ContinentPage">I am an ContinentPage</div>;
  }
}

export const mapStateToProps = store => {
  return {
    animals: store.animals,
    continents: store.continents
  };
};

export default connect(mapStateToProps, null)(ContinentPage);
