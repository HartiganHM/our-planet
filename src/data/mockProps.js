import mockAnimalsArray from './mockAnimalsArray';
import mockContinentsArray from './mockContinentsArray';

const mockProps = {
  match: { params: { animal: 'Arctic Fox', continent: 'Africa' } },
  animals: mockAnimalsArray,
  continents: mockContinentsArray
};

export default mockProps;
