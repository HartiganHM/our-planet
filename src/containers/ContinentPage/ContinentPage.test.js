import React from 'react';
import { shallow } from 'enzyme';
import { ContinentPage, mapStateToProps } from './ContinentPage';
import mockAnimalsArray from '../../data/mockAnimalsArray';
import mockContinentsArray from '../../data/mockContinentsArray';
import mockProps from '../../data/mockProps';

describe('ContinentPage tests', () => {
  let renderedContinentPage;

  beforeEach(() => {
    renderedContinentPage = shallow(
      <ContinentPage
        animals={mockProps.animals}
        continents={mockProps.continents}
        match={mockProps.match}
      />
    );
  });

  it('Should match the snapshot', () => {
    expect(renderedContinentPage).toMatchSnapshot();
  });

  it('matchContinentAnimals should match animals to a given continent id', () => {
    const expected = mockAnimalsArray;
    const mockContinentId = mockContinentsArray[1].id;
    const result = renderedContinentPage
      .instance()
      .matchContinentAnimals(mockContinentId);

    expect(result).toEqual(expected);
  });

  it('matchContinentAnimals returns an empty array if continent id does not match', () => {
    const expected = [];
    const mockContinentId = mockContinentsArray[0].id;
    const result = renderedContinentPage
      .instance()
      .matchContinentAnimals(mockContinentId);

    expect(result).toEqual(expected);
  });

  describe('mapStateToProps tests', () => {
    it('Should pull animals from the store', () => {
      const mockStore = {
        animals: mockProps.animals
      };
      const result = mapStateToProps(mockStore);

      expect(result.animals).toEqual(mockStore.animals);
    });

    it('Should pull continents from the store', () => {
      const mockStore = {
        continents: mockProps.continents
      };
      const result = mapStateToProps(mockStore);

      expect(result.continents).toEqual(mockStore.continents);
    });
  });
});
