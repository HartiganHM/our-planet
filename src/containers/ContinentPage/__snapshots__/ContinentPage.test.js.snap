// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContinentPage tests Should match the snapshot 1`] = `ShallowWrapper {}`;
